### PR TITLE
Removes camera assembly structures

### LINKED ISSUE #79019: Cameras created from camera assemblies accept upgrades but don't upgrade
## Reproduction:

1. Make a camera assembly (autolathe)
2. Create a camera
3. Add any upgrade (gas analyzer, for example) and watch it get ate. The prompt "the camera has been upgraded.." still fires, but the actual upgrade doesn't. Upgrades tested were prox sensor and gas analyzer.

Anyways. I was interested in fixing this issue myself, but I also want to properly document it.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Removes the camera assembly structure middleman between the camera wallframe and camera machine. All its behavior has been instead moved to the camera, and I've tried to keep as much of the behavior the same as before.
This also fixes the issue that camera assemblies had where, upon the construction being finished, it would move itself into the newly finished camera machine, therefore taking itself off a wall, therefore deconstructing itself. This resulted in 2 piece of iron being in each camera machine (except roundstart ones), and because camera machines rely on the assembly inside of them for upgrades and such, upgrading didn't work at all.

I've also made camera nets use defines (not in map) so it's easier to find a list of them all, and tried to add autodoc comments to nearly every var in camera code.

## Why It's Good For The Game

Removes copy paste and spaghetti code between structure and machine camera, thus making it easier to work around with.
Closes https://github.com/tgstation/tgstation/issues/79019

## Changelog

:cl:
fix: Cameras built in-round can be upgraded again.
fix: Deconstructing cameras now more consistently return to you the upgrades inside of the camera.
fix: RD's telescreen can now properly see Ordnance cameras again.
fix: [Deltastation] Library art gallery no longer has an invisible camera.
/:cl:

### CODE REVIEW

File: code/game/machinery/camera/camera.dm [hunk 1]
```diff
@@ -74,6 +74,11 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/camera/xray, 0)
 	if(old_assembly) //check to see if the camera assembly was upgraded at all.
 		assembly = old_assembly
 		assembly_ref = WEAKREF(assembly) //important to do this now since upgrades call back to the assembly_ref
+		setDir(assembly.dir)
+		assembly.forceMove(src)
+		var/area/A = get_area(src)
+		c_tag = "[format_text(A.name)] ([rand(1, 999)])"
```
Comment (san7890) [inline code/game/machinery/camera/camera.dm:80]: let's yeet the single letter variable while we're here (also you already did this in another place)

```suggestion
		var/area/assembly_area = get_area(src)
		c_tag = "[format_text(assembly_area.name)] ([rand(1, 999)])"
```

**Review by san7890 — DISMISSED** [review]

**Comment by MrMelbert** [comment]
Does this keep the wall safe hanging in the air then? If so this is not a good fix https://github.com/tgstation/tgstation/pull/80589#issuecomment-1869815715

**Comment by JohnFulpWillard** [comment]
The only other thing I can think of would be to remove it as being a wall item entirely. Even as a wall item, you can incinerate/SM it and destroy the one spare gold ID in the game

**Comment by MrMelbert** [comment]
In any case, DNM while it leaves stuff hanging in the air

**Comment by MrMelbert** [comment]
> Even as a wall item, you can incinerate/SM it and destroy the one spare gold ID in the game

Also I think this is fine
- Most of the time (barring lowpop) the spare is not in the safe beyond the first 10 minutes
- Most people would rather steal the spare ID rather than destroy it
- Nothing is gained from going out of your way to chuck the safe into the SM
- Even if it's destroyed, it's not like its existence is critical for game progression: the shuttle can still be called assuming there's any head of staff or any crewmember with ID console access or an AI or through destroying all the consoles or a restart vote or an admin calling it
- At worst, an admin could send a spare spare, maybe with a centcom inspector, make it an event.

**Comment by JohnFulpWillard** [comment]
I'll just leave for another PR then

File: code/datums/components/wall_mounted.dm [hunk 2]
```diff
@@ -71,6 +71,9 @@
 	SIGNAL_HANDLER
 	var/obj/hanging_parent = parent
 
+	if(hanging_parent.obj_flags & NO_DECONSTRUCTION)
+		return
```
Comment (MrMelbert) [inline code/datums/components/wall_mounted.dm:75]: This still needs to be reverted

...Ideally the `.deconstruct` call would handle `NO_DECONSTRUCTION` handling as it wants, but there's a shocking lack of consistency around `NO_DECONSTRUCTION`, where some objects treat it as "allow deconstruction, but make it drop no parts" and others simply "disallow deconstruction at all"

File: code/__DEFINES/cameras.dm [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#define CAMERA_STATE_WRENCHED 1
+#define CAMERA_STATE_WELDED 2
+#define CAMERA_STATE_WIRED 3
+#define CAMERA_STATE_FINISHED 4
```
Comment (MrMelbert) [inline code/__DEFINES/cameras.dm:4]: Put this in the construction subfolder of defines

File: code/datums/components/wall_mounted.dm (hunk 2 above)
Comment (JohnFulpWillard) [reply code/datums/components/wall_mounted.dm:75]: This is what's fixing the cameras, if something has ``NO_DECONSTRUCTION`` flag then we shouldn't let it be deconstructed from it falling off of the wall. If something is left hanging over nothing then we should fix it case-by-case, if something is set to no_deconstruct then it should not deconstruct.

Comment (MrMelbert) [reply code/datums/components/wall_mounted.dm:75]: Then you disagree with a lot of assumptions made by the code base, which implies this isn't a good fix (at least, not right now)

Comment (JohnFulpWillard) [reply code/datums/components/wall_mounted.dm:75]: What do you suggest would be a better fix here because I don't really agree.

**Pull request merged**